lib/python/unit.cpp

```diff
@@ -17,6 +22,8 @@ bool temporal_or_dimensionless(const units::Unit unit) {
 } // namespace
 
 units::Unit make_unit(const ProtoUnit &unit) {
+  if (std::holds_alternative<py::none>(unit))
+    return units::none;
   if (std::holds_alternative<std::string>(unit))
     return units::Unit(std::get<std::string>(unit));
   return std::get<units::Unit>(unit);
```

Comment:
Thinking about it, it is not about usefulness, but I think it is actually risking a segfault (because `make_unit` will use `std::get<Unit>` even though the stored alternative may be `DefaultUnit`), so I will fix.